gcc/rust/util/rust-optional.h

```diff
@@ -86,6 +86,7 @@ template <typename T> class Optional
   union
   {
     T value;
+    volatile char unused;
```

Comment:
Please add a comment explaning the fix and linking the issue you put in your github comment. We also need a `FIXME: Remove this` so we remember to get to it